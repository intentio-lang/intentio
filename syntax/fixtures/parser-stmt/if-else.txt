if a {
    if b {
        c;
    } else {
        d;
    }
} else if e {
    f;
}
