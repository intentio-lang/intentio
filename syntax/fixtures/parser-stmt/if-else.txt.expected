[
  {
    "_stmtAnn": [],
    "_stmtKind": {
      "contents": {
        "_exprAnn": [],
        "_exprKind": {
          "contents": [
            {
              "_exprAnn": [],
              "_exprKind": {
                "contents": {
                  "contents": {
                    "_scopeIdAnn": [],
                    "_scopeIdSourcePos": [
                      "test.ieo",
                      0,
                      0
                    ],
                    "_unScopeId": "a"
                  },
                  "tag": "ScopeId'"
                },
                "tag": "IdExpr"
              },
              "_exprSourcePos": [
                "test.ieo",
                0,
                0
              ]
            },
            {
              "_blockAnn": [],
              "_blockSourcePos": [
                "test.ieo",
                0,
                0
              ],
              "_blockStmts": [
                {
                  "_stmtAnn": [],
                  "_stmtKind": {
                    "contents": {
                      "_exprAnn": [],
                      "_exprKind": {
                        "contents": [
                          {
                            "_exprAnn": [],
                            "_exprKind": {
                              "contents": {
                                "contents": {
                                  "_scopeIdAnn": [],
                                  "_scopeIdSourcePos": [
                                    "test.ieo",
                                    0,
                                    0
                                  ],
                                  "_unScopeId": "b"
                                },
                                "tag": "ScopeId'"
                              },
                              "tag": "IdExpr"
                            },
                            "_exprSourcePos": [
                              "test.ieo",
                              0,
                              0
                            ]
                          },
                          {
                            "_blockAnn": [],
                            "_blockSourcePos": [
                              "test.ieo",
                              0,
                              0
                            ],
                            "_blockStmts": [
                              {
                                "_stmtAnn": [],
                                "_stmtKind": {
                                  "contents": {
                                    "_exprAnn": [],
                                    "_exprKind": {
                                      "contents": {
                                        "contents": {
                                          "_scopeIdAnn": [],
                                          "_scopeIdSourcePos": [
                                            "test.ieo",
                                            0,
                                            0
                                          ],
                                          "_unScopeId": "c"
                                        },
                                        "tag": "ScopeId'"
                                      },
                                      "tag": "IdExpr"
                                    },
                                    "_exprSourcePos": [
                                      "test.ieo",
                                      0,
                                      0
                                    ]
                                  },
                                  "tag": "ExprStmt"
                                },
                                "_stmtSourcePos": [
                                  "test.ieo",
                                  0,
                                  0
                                ]
                              }
                            ]
                          },
                          {
                            "_blockAnn": [],
                            "_blockSourcePos": [
                              "test.ieo",
                              0,
                              0
                            ],
                            "_blockStmts": [
                              {
                                "_stmtAnn": [],
                                "_stmtKind": {
                                  "contents": {
                                    "_exprAnn": [],
                                    "_exprKind": {
                                      "contents": {
                                        "contents": {
                                          "_scopeIdAnn": [],
                                          "_scopeIdSourcePos": [
                                            "test.ieo",
                                            0,
                                            0
                                          ],
                                          "_unScopeId": "d"
                                        },
                                        "tag": "ScopeId'"
                                      },
                                      "tag": "IdExpr"
                                    },
                                    "_exprSourcePos": [
                                      "test.ieo",
                                      0,
                                      0
                                    ]
                                  },
                                  "tag": "ExprStmt"
                                },
                                "_stmtSourcePos": [
                                  "test.ieo",
                                  0,
                                  0
                                ]
                              }
                            ]
                          }
                        ],
                        "tag": "IfExpr"
                      },
                      "_exprSourcePos": [
                        "test.ieo",
                        0,
                        0
                      ]
                    },
                    "tag": "ExprStmt"
                  },
                  "_stmtSourcePos": [
                    "test.ieo",
                    0,
                    0
                  ]
                }
              ]
            },
            null
          ],
          "tag": "IfExpr"
        },
        "_exprSourcePos": [
          "test.ieo",
          0,
          0
        ]
      },
      "tag": "ExprStmt"
    },
    "_stmtSourcePos": [
      "test.ieo",
      0,
      0
    ]
  }
]
